### PR TITLE
New ros example

### PR DESCRIPTION
**Pull request by adipokala**
Please, review the example and let me know if my understanding is correct.

Will squash / rebase and merge.

### CODE REVIEW

**Review by mharshe — CHANGES_REQUESTED** [review]

File: tutorials/chapter2.md [hunk 1]
```diff
@@ -274,6 +274,105 @@ int main(int argc, char **argv) {
 * The Environment class interface can be used to get and set values. When getting/setting a parameter, you must know the type of the parameter and use the relevant getter/setter. 
 * After running this example, you can verify that a new parameter `/paramFloat` exists with a value 1.5 using the `rosparam` command-line utility.
 
+#### Example 5: Exchange ROS Messages
+
+This example shows us that with Klepsydra core you can use ROS messages to exchange data between the nodes.
```
Comment (mharshe) [inline tutorials/chapter2.md:279]: Add more explanation about what is the difference between this example and the others.

File: tutorials/chapter2.md [hunk 2]
```diff
@@ -274,6 +274,105 @@ int main(int argc, char **argv) {
...
+{
+    std::string topicName1 = "tutorial_api_example5_1";
+    ros::Publisher strPublisher = nodeHandle.advertise<std_msgs::String>(topicName1, 20);
+    kpsr::ros_mdlw::FromRosMiddlewareProvider fromRosProvider(nodeHandle);
+    // Declare the eventloop and vector publisher
+    kpsr::high_performance::EventLoopMiddlewareProvider<16> eventloop(nullptr);
+
+    eventloop.start();
+
+    // auto strpublisher = eventloop.getPublisher<std_msgs::String>("example5", 0, nullptr, nullptr);
```
Comment (mharshe) [inline tutorials/chapter2.md:306]: in fact, the idea is to use kpsrPublisher to publish data. This way, any class in the rest of your code that uses Publisher* can still work and will in fact use the underlying ros publisher to actually transmit data.

For example, the SimplePublisher class in these tutorial examples has a Publisher pointer. So you could use the SimplePublisher class with these ros hooks and now without changing your original class you have added ROS support. the only thing that changes is your calls in the main() and your original SimplePublisher is agnostic to ros.

**Review by mharshe — COMMENTED** [review]
What the examples in the tutorials are trying to show is that you could develop your application with pub/sub architecture that only use Klepsydra Publisher* and Subscriber* and the depending on the application add ROS support for transmitting messages without changing your application. Only the part where we set up and connect the elements will have ros specific elements.

However, many real world applications make heavy use ROS message types and for these applications we want to avoid the overhead of converting data types. The purpose of these examples is to show that in fact with Klepsydra you can do that --- you can have a loose link with ROS where we have automatic conversion from standard data types to ROS messages, or you can have a deeper dependency to ROS and use the ros message types without having to deal with overhead of conversion.

So update the tutorial text to reflect this and change the rosToKpsr amd kpsrToRos functions too.

**Review by pablogh-2000 — COMMENTED** [review]
@adipokala please respond to all comments with either:
- Done (if you did code changes in response to the comment)
- Explanation (if you did no code changes in response to the comment for some reason)

**Review by pablogh-2000 — APPROVED** [review]

**Pull request merged**